### ISSUE TITLE
Translate game to Norwegian Bokmål (nb)?

### ISSUE THREAD

**Issue by petterreinholdtsen** [openclonk/openclonk#81]
Hi

What is needed to translate this game to Norwegian Bokmål?  Please consider making translation available via web services like <URL: https://hosted.weblate.org >.

**Comment by gitMarky** [comment]
Hi,

first of all, we would need someone who can translate from English or German to Norwegian. Funny enough, I learned some Norwegian not long ago, but I don't have enough knowledge to translate the entire game, nor do I have the motivation (because we do not even know how many players could benefit from this!). 

If you want to try it yourself you just have to find all the text files that contain localization, then add the Norwegian translation. See https://docs.openclonk.org/en/sdk/lang.html

**Comment by petterreinholdtsen** [comment]
[gitMarky]
> If you want to try it yourself you just have to find all the text
> files that contain localization, then add the Norwegian
> translation. See https://docs.openclonk.org/en/sdk/lang.html

Thank you for the link.  Did you consider to collect the all in a pot
file to make it easier for translators?

-- 
Happy hacking
Petter Reinholdtsen

**Comment by 772** [comment]
https://github.com/openclonk/openclonk/blob/master/planet/System.ocg/LanguageUS.txt contains all the texts visible in the main menu etc. If you like to start translating the game, I would suggest to translate that one file. The scenario and object descriptions and names are more likely to change in the future. As soon as you place a new LanguageNO.txt in the System.ocg, you will be able to choose Norwegian in the game options. (Similar request: https://forum.openclonk.org/topic_show.pl?pid=29035 )

And what's even more important: Please tell us if there is any Norwegian letter missing in the Endeavour.ttf (standard font size).

**Comment by petterreinholdtsen** [comment]
I notice the LanguageUS.txt content look like something the po4a ini file format should be able to handle, if quotes are added to the file.  If you go this way, you can generate a pot file that can be provided to translators using any po file editor, and even translated using web based translation portals like weblate.

**Comment by 772** [comment]
Using .po files the way that LibreOffice, GIMP, Filezilla and most other multilingual open source projects use them seems is probably a good idea, yes. A few years ago, this was maybe not discussed because scenario descriptions were .rtf files and not .txt. The engine translation is a different thing anyway but still, rtf-files were a bigger (more important) mess imo.

Opinions on integrating .po-files in OC?

**Comment by gitMarky** [comment]
It may be a good idea, but I looked at the format and I do not really like it, mostly because you get a lot of duplicate text? On the other hand, I did not work with them yet, so these are only assumptions. See https://www.gnu.org/software/gettext/manual/html_node/PO-Files.html

In our case this might be:

> Log("$MyText$");
> with a translation 
> #: MyObject.ocd/Script.c:123
> msgid "$MyText$"
> msgstr "This is my text, read it please"

Or, if we stop using the $$-format:

> Log("This is my text, read it please");
> with a translation 
> #: MyObject.ocd/Script.c:123
> msgid "This is my text, read it please"
> msgstr "Das ist mein Text, bitte lesen"

No idea how well that works. It may be convenient for the translator when they see the text right away, but it seems to be auto-generated, or otherwise I cannot imagine how you'd keep up with the references if there are code changes, or if the text for msgid changes.

**Comment by mikaelssen** [comment]
I'll get on this. I'm native Norwegian speaker, A list of language files would be appriciated though 👍

**Comment by petterreinholdtsen** [comment]
Yes, there are automatic tools available to extract strings and generate pot files, and to merge POT and PO files together when the original strings change.

A starting point can be https://www.gnu.org/software/gettext/, as gettext is the original implementation using POT and PO files.
